spyder/widgets/ipythonconsole/shell.py

```diff
@@ -243,11 +243,11 @@ def handle_exec_method(self, msg):
                         self.sig_namespace_view.emit(view)
                     else:
                         view = None
-                elif 'get_var_properties' in method:
+                elif 'get_var_properties' in method and 'text/plain' in data:
```

Comment:
Also, please change the above `if` clause to be

```python
if 'get_namespace_view' in method:
    if data is not None and 'text/plain' in data:
        view = ast.literal_eval(data['text/plain'])
    else:
        view = None
    self.sig_namespace_view.emit(view)
```

That way we always emit a signal there.